packages/accounts/plugingen/phases/plugin-actions/index.ts

```diff
@@ -87,9 +87,9 @@ export const PluginActionsGenPhase: Phase = async (input) => {
 
   addType(
     `ExecutionActions<
-      TEntryPointVersion extends EntryPointVersion,
-      TAccount extends SmartContractAccount<TEntryPointVersion> | undefined =
-        SmartContractAccount<TEntryPointVersion> | undefined
+      TAccount extends SmartContractAccount | undefined =
+        SmartContractAccount | undefined,
+      TEntryPointVersion extends GetEntryPointFromAccount<TAccount> = GetEntryPointFromAccount<TAccount>
```

Comment:
I think it's better to not infer from account for object type definitions, and I only do it for function type definitions.

The reason is the dev can still pass in (or have to pass in) TEntryPointVersion themselves to the type and we don't want to assume that the actual entrypointversion of account equals the following entry point version type.

For function types, such case could be handled robustly and gracefully in run-time. For type definitions, I rather prefer explicit type definitions by the developers "if" they want to type it.

By having `TEntryPointVersion extends EntryPointVersion,` do be explicit set by the dev, and no infer fallback, we can for sure assume that account entry point matches the type.

Reply:
I disagree. The methods here should not be allowed to operate on EP versions other than the ones that the connected account supports. Otherwise someone might use EP 0.7 on an account that doesn't support it. 

The EP version is ultimately tightly coupled with the smart contract account so it should be a limited set. This set up of inferring the EP, but still allowing the generic to be passed in still allows for overriding behaviour